runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -3604,6 +3608,25 @@ J9::ValuePropagation::getParmValues()
          bool isClassErased = false;
 
          TR_OpaqueClassBlock *opaqueClass = parmIterator->getOpaqueClass();
+
+         if (opaqueClass &&
+             parmIterator->isArray() &&
+             TR::Compiler->om.areFlattenableValueTypesEnabled())
+            {
+            TR_OpaqueClassBlock *arrayComponentClass = comp()->fej9()->getComponentClassFromArrayClass(opaqueClass);
+            uint32_t len = 0;
+
+            // Both regular nullable array and null-restricted array have the same signature.
+            // If the array component class is not an identity type, the array might either be
+            // a null-restricted array or a nullable array. Therefore, we can't trust the array
+            // class returned by the signature.
+            if (!TR::Compiler->cls.classHasIdentity(arrayComponentClass) ||
+                !strncmp(parmIterator->getUnresolvedJavaClassSignature(len), "[Ljava/lang/Object;", 19))
```

Comment:
If `len` is less than 19, this will compare beyond the end of the class signature.  Although it's not likely to cause a problem, I think it would be safer to do something like this:

```
char *classSig = parmIterator->getUnresolvedJavaClassSignature(len);
if (!TR::Compiler->cls.classHasIdentity(arrayComponentClass) ||
    (len == 19 && !strncmp(classSig, "[Ljava/lang/Object;", 19)))
```

Reply:
Addressed in [cc811af](https://github.com/eclipse-openj9/openj9/compare/6782be9764d21471a7b2d4a16e71519dbe36ceba..cc811af332be4d1aaabc0318586abad687da9eb0)